tests/unit/test_energy_spectrum.py

```diff
@@ -271,5 +255,181 @@ def test_when_no_foil_is_true(self):
+                (3.75, 0.0)]
+
+    @property
+    def new_spectra(self):
+        return {'1H.ERD.2': [
+            (-0.95, 0), (-0.45, 0.0), (0.45, 7.0), (0.95, 1.0), (1.45, 0)]
+        }
+
+    @property
+    def expected_sum_spectrum_file_content(self):
+
+        return [
+            '-0.75000      0',
+            '-0.25000      0',
+            '0.25000      7',
+            '0.75000     11',
+            '1.25000      0',
+            '1.75000      2',
+            '2.25000      8',
+            '2.75000      3',
+            '3.25000      6',
+            '3.75000      0'
+        ]
+
+    @property
+    def get_energy_spectra(self):
+        return {
+            '1H.ERD.0': [
+                (-0.25, 0), (0.25, 0.0), (0.75, 10.0), (1.25, 0)
+            ],
+            '1H.ERD.1': [
+                (-0.75, 0), (-0.25, 0.0), (0.25, 7.0), (0.75, 1.0), (1.25, 0)
+            ],
+            '35Cl.RBS_Mn.0': [
+                (1.75, 0), (2.25, 0.0), (2.75, 3.0), (3.25, 6.0), (3.75, 0)
+            ],
+            '7Li.0.0.0': [
+                (0.75, 0), (1.25, 0.0), (1.75, 2.0), (2.25, 8.0), (2.75, 0)
+            ]
+        }
+
+    def test_sum_spectrum_creation(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+
+            sum_spectrum, es, _ = self.create_sum_and_energy_spectrum(tmp_dir)
+
+            # Check if a created sum spectrum is correct
+            self.assertEqual(sum_spectrum.spectra, es)
+            self.assertEqual(sum_spectrum.sum_spectrum,
+                             self.created_sum_spectrum)
+        return
+
+    def test_sum_spectrum_removal(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sum_spectrum, es, _ = self.create_sum_and_energy_spectrum(tmp_dir)
+
+            removed_spectrum = {
+                "1H.ERD.0": es["1H.ERD.0"]
+            }
+
+            # Check if a deleted spectrum is really removed
+            sum_spectrum.delete_spectra(removed_spectrum)
+            self.assertEqual(sum_spectrum.sum_spectrum,
+                             self.sum_spectrum_after_removed_spectra)
+        return
+
+    def test_sum_spectra_addition(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sum_spectrum, es, _ = self.create_sum_and_energy_spectrum(tmp_dir)
+
+            sum_spectrum.add_or_update_spectra(self.new_spectra)
+
+            # Check if an added spectra is really added
+            self.assertEqual(sum_spectrum.sum_spectrum,
+                             self.sum_spectrum_with_added_spectra)
+        return
+
+    def test_empty_sum_spectrum(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sum_spectrum, es, mesu = self.create_sum_and_energy_spectrum(
+                tmp_dir)
+
+            empty_spectrum = SumEnergySpectrum(None,
+                                               mesu.get_energy_spectra_dir(),
+                                               SumSpectrumType.MEASURED)
```

Comment:
Here's where the combined helper gets in the way. The SumEnergySpectrum is created once in the helper and another time in the test.